beard/metrics/text.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Beard.
+# Copyright (C) 2015 CERN.
+#
+# Beard is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""String metrics.
+
+.. codeauthor:: Petros Ioannidis <petros.ioannidis91@gmail.com>
+
+"""
+
+from __future__ import division
+import re
+
+
+def find_all(s, pattern):
+    """Find all occurences of the given pattern.
+
+    Parameters
+    ----------
+    :param s: string
+        String to be searched
+
+    :param letter: string
+        Substring we are searching for
+
+    Returns
+    -------
+    :returns: generator
+        A generator that holds the indexes of the patterns
+    """
+    for match in re.finditer(pattern, s):
+        yield match.start()
+
+
+def jaro_matching(s1, s2):
```

Comment:
Can you rename this function? Since it now returns both the number of matching letters and number of transpositions. Please update the docstring as well.